### PR TITLE
feat: Add support for product export in UI

### PR DESCRIPTION
**Pull request by sradevski**
I've keep the PR simple, so it is missing column selection (we should discuss if we event want it now or in the polish stage) and push notifications (similarly, needs discussion if we want to tackle it right away).

One thing we need to improve (outside of the scope here) is to preserve query params on `handleSuccess` call, so filters and such is not lost if you eg. go to `create` and go back.

CLOSES CC-234

### CODE REVIEW

File: packages/admin-next/dashboard/src/components/table/data-table/data-table-filter/date-filter.tsx [hunk 1]
```diff
@@ -118,123 +119,136 @@ export const DateFilter = ({
...
-                "filters_menu_content"
-              ) {
-                e.preventDefault()
+      <DateDisplay
+        label={label}
+        value={displayValue}
+        onRemove={handleRemove}
+        readonly={readonly}
+      />
+      {!readonly && (
```
Comment (sradevski) [inline packages/admin-next/dashboard/src/components/table/data-table/data-table-filter/date-filter.tsx:128]: These are just indentation changes, no actual code changed other than the readonly check.

File: packages/admin-next/dashboard/src/components/table/data-table/data-table-filter/data-table-filter.tsx [hunk 2]
```diff
@@ -41,10 +41,15 @@ export type Filter = {
...
 
-export const DataTableFilter = ({ filters, prefix }: DataTableFilterProps) => {
+export const DataTableFilter = ({
+  filters,
+  readonly,
+  prefix,
+}: DataTableFilterProps) => {
   const { t } = useTranslation()
   const [searchParams] = useSearchParams()
   const [open, setOpen] = useState(false)
```
Comment (sradevski) [inline packages/admin-next/dashboard/src/components/table/data-table/data-table-filter/data-table-filter.tsx:55]: Question: Can't we rely solely on the query parameters for the `activeFilters` state? I feel like it should be possible, and we can remove some code around maintaining their state (so basically we have getInitialFilters be the state, and that gets memoized with `location.search` passed to it)

**Comment by olivermrbl** [comment]
@sradevski, this doesn't seem to work-out-of-the box with the local file provider – did you use S3? 

The exported CSV file is put into the `/private` folder, which is not served by the server as far as I can see. To be downloadable from admin, this should be placed in static no?

Update: Perhaps the issue is also related to the fact that the generated URL of the exported file points to a path on the local file system and not one on the server. Not sure if our Vite application can deal with this?

**Comment by sradevski** [comment]
@olivermrbl That behavior is intentional actually. The problem with using the local file provider for the exports is that we can't have security controls (so, no support for private files) if we put it in static, as anyone would be able to read the file, so I create a `private` folder where private files would go (and you would need direct access to the machine to fetch them).

Let me know if you think this should behave differently, it's super easy to change it.

**Comment by olivermrbl** [comment]
> The problem with using the local file provider for the exports is that we can't have security controls (so, no support for private files) if we put it in static, as anyone would be able to read the file

Given the local file provider is only used for development, wouldn't this be OK?

**Comment by sradevski** [comment]
> > The problem with using the local file provider for the exports is that we can't have security controls (so, no support for private files) if we put it in static, as anyone would be able to read the file
> 
> Given the local file provider is only used for development, wouldn't this be OK?

I've heard people use it in production as well for some small sites, so I thought to err on the safe side. If we are OK with it since people really shouldn't use it in prod, then we can def. put it in `static`. WDYT?

You can actually configure if you want it in static or private currently (you just need to set `privateUploadDir_` to the same URL as `uploadDir_`), but it defaults to private. Maybe we can default to `static` instead?

**Comment by olivermrbl** [comment]
> You can actually configure if you want it in static or private currently (you just need to set privateUploadDir_ to the same URL as uploadDir_), but it defaults to private. Maybe we can default to static instead?

Originally (in V1), the provider was supposed to be for development only, and IMO, this should still be the recommendation from our side. With that in mind, I would suggest we optimize for this use case such that we can guarantee a good first experience across all our features. 

Then in the rare (my hunch; how often have you heard this being used in prod?) case, we can guide developers to a secure configuration with the local provider.

Wdyt?

**Comment by sradevski** [comment]
@olivermrbl it should definitely be only for development, but I've seen at least 2 conversations on Discord, and one from a local agency (and I obviously told them to use S3 😄).

Then let me use the same, static directory for private files as well, and leave a comment that if they really wanted to they can change it through the config, WDYT?

**Comment by olivermrbl** [comment]
> Then let me use the same, static directory for private files as well, and leave a comment that if they really wanted to they can change it through the config, WDYT?

Sounds good!

**Comment by olivermrbl** [comment]
@sradevski, the static folder is correctly used now for the file uploads. However, we still have the issue with the admin dashboard attempting to download from the local file system. This leads to the download containing the `index.html` because that's the fallback in case the requested file cannot be found.

**Comment by olivermrbl** [comment]
@kasperkristensen, could you give this a review? 🙏

**Comment by sradevski** [comment]
@olivermrbl I missed removing one check (see last commit), should be good now.

**Review by olivermrbl — APPROVED** [review]
LGTM!

**Comment by olivermrbl** [comment]
@sradevski, works nicely – good one 👍

**Pull request merged**